### ISSUE TITLE
Incorrectly reported missing `clone` method

### ISSUE THREAD

**Issue by rachitnigam** [rust-lang/rust-analyzer#14308]
**rust-analyzer version**: rust-analyzer version: 0.3.1426-standalone (0a956ec93 2023-03-05)
**rustc version**: rustc 1.68.0 (2c8cc3432 2023-03-06)
**relevant settings**: Check on save command is `cargo clippy` but error also occurs when configured with `cargo check` 

The VS code extension for rust-analyzer reports that clone is missing on for a struct:
<img width="679" alt="Screenshot 2023-03-09 at 3 57 26 PM" src="https://user-images.githubusercontent.com/15348827/224187999-451ad673-5695-4450-8266-2dcbe9fe48b0.png">

However, the struct derives Clone:
<img width="260" alt="Screenshot 2023-03-09 at 3 57 58 PM" src="https://user-images.githubusercontent.com/15348827/224188067-9d32c70b-53a7-452a-8d90-308db9ce0584.png">

More importantly, running `cargo build` or `cargo check` from the command line doesn't give any error. I'm not sure what's causing the error but it did randomly show up recently. I'm assuming it was caused by an updated.

The only workaround I've found is replacing `struct.clone` with `Clone::clone(&struct)`:
<img width="297" alt="Screenshot 2023-03-09 at 3 59 19 PM" src="https://user-images.githubusercontent.com/15348827/224188263-97f672e9-f069-4820-8da2-61576afc9c44.png">

To reproduce:
```
git clone https://github.com/cucapra/filament.git
cd filament && code .
```

Happy to run more debugging commands to help figure out what's causing the problem.

**Comment by rachitnigam** [comment]
I can confirm that switching to `0.3.1418-standalone (4e29820f6 2023-02-26)` fixes the problem for me.

**Comment by lnicola** [comment]
The warnings will be disabled by default in the next version, but it would be interesting to figure out what's happening here. The type there is barely a:

```rust
#[derive(Clone)]
pub struct SExp(pub String);
```

For future readers, you'll also need to clone git clone https://github.com/cucapra/calyx.git.

**Comment by rachitnigam** [comment]
Got it! I'd be interested in poking around the codebase if you have any pointers on where I should look?

**Comment by lnicola** [comment]
Um, no idea. First step might be to minimize this, but I'm currently running into #14310, which may have some impact on it.

**Comment by lnicola** [comment]
Duplicate of #13600.
